src/replacements.js

```diff
@@ -0,0 +1,188 @@
+const Gio = imports.gi.Gio;
+const EosGates = imports.eos_gates;
+
+function generateEndlessInstallerEntry() {
+    let entry = {
+        regex: {
+            windows: /endless-installer.*.exe/
+        },
+        overrideHelpMessage: _("You are already running Endless OS")
+    };
+
+    // If we detect the presence of eos-installer.desktop, inform
+    // the user on how to actually install Endless OS.
+    let desktopInfo = Gio.DesktopAppInfo.new('eos-installer.desktop');
+    if (desktopInfo) {
+        entry.overrideHelpMessage = _("You are already running Endless OS from live media");
+        entry.desktopInfo = desktopInfo,
+        entry.replacementInfo = {
+            appName: _("Endless OS Installer"),
+            description: _("You can install Endless OS by running the Endless OS Installer.")
+        }
+    }
+
+    return entry;
+}
+
+function definitions() {
+    return [
+        {
+            regex: {
+                windows: /spotifysetup.*.exe/i
+            },
+            appName: _("Spotify"),
+            flatpakInfo: { remote: 'eos-apps', id: 'com.spotify.Client' }
```

Comment:
Damn, okay, this is not something that we can get from Flatpak. `flatpak_installation_get_installed_ref` only works for installed apps and I need a `FlatpakInstlaledRef` in order to use `get_origin()`. It looks like we'll need to specify the remote explicitly here for now (would be nice as you say to be able to get this dynamically).

I believe the reason why GS requires the 'fully resolved' ID (including the remote name) is because an app-id like `com.spotify.Client` is ambiguous when GS has to support more than just Flatpak.